src/spec.ts

```diff
@@ -225,89 +211,75 @@ export function normalizeUnitSpec(spec: UnitSpec): Spec {
     (overlayConfig.area === 'linepoint' && spec.mark === AREA)
   );
 
-  // TODO: thoroughly test
-  if (spec.mark === ERRORBAR) {
-    return normalizeErrorBarUnitSpec(spec);
-  }
-  // TODO: thoroughly test
-  if (isRanged(spec.encoding)) {
-    return normalizeRangedUnitSpec(spec);
-  }
+  if (isNormalUnitSpec(spec)) {
+    // TODO: thoroughly test
+    if (isRanged(spec.encoding)) {
+      return normalizeRangedUnit(spec);
+    }
 
-  if (overlayWithPoint || overlayWithLine) {
-    return normalizeOverlay(spec, overlayWithPoint, overlayWithLine);
+    if (overlayWithPoint || overlayWithLine) {
+      return normalizeOverlay(spec, overlayWithPoint, overlayWithLine);
+    }
+    return spec;
+  } else {
+    /* istanbul ignore else */
+    if (spec.mark === ERRORBAR) {
+      return normalizeErrorBar(spec);
+    } else {
+      throw `unsupported composite mark ${spec.mark}`;
```

Comment:
Actually, throwing a string is discouraged. https://stackoverflow.com/questions/11502052/throwing-strings-instead-of-errors

Better `throw new Error(message)`.